src/routes/search/containers/SearchPage.js

```diff
@@ -32,15 +32,27 @@ export class SearchPage extends React.Component {
   }
 
   acceptCategory (category) {
-    const title = category.title.toLowerCase()
-    const content = category.content
     const filterText = this.state.filterText.toLowerCase()
-    return title.includes(filterText) || content.toLowerCase().includes(filterText)
+    const title = category.title.toLowerCase()
+    const content = category.content.toLowerCase()
+    return title.includes(filterText) || content.includes(filterText)
   }
 
   findCategories () {
+    const filterText = this.state.filterText.toLowerCase()
+
     return this.props.categories.toArray()
       .filter(category => this.acceptCategory(category))
+      // sort results so that categories including the filter text in the title on top
+      .sort((category1, category2) => {
```

Comment:
Should we maybe also sort alphabetically? Also I trust that code that it sorts correctly but it's rater complicated imo. Maybe some comments would be nice.

Reply:
we could also filter for a match in title and a match in content separately and concat the results afterwards if that increases readability? alphabetical sorting is okay for me, but adds more complexity

Reply:
We can sort first alphabetically and afterwards by an occurance by title.

Reply:
I wouldn't bother alphabetical sorting... Doesn't make sense in a search function and nobody expects alphabetical sorting there...
Maybe a better-readable version would be:
```javascript
prioritizeCategory (category) {
  const filterText = this.state.filterText.toLowerCase()
  if (category.title.toLowerCase().includes(filterText)) {
    return 2
  } else if (category.content.toLowerCase().includes(filterText)) {
    return 1
  } else {
    return 0
  }
}

findCategories () {
  return this.props.categories.toArray()
    .map(category => ({priority: prioritizeCategory(category), category})
    .filter(({priority, category}) => priority !== 0)
    .sort((wrapper1, wrapper2) => wrapper1.priority - wrapper2.priority)
    .map(wrapper => ({model: wrapper.category, children: []))
}
```

Reply:
I think alphabetical sorting makes sense because imo it would be nice if the search is deterministic.

Reply:
I also like alphabetical sorting :D

Reply:
@Schedulaar your suggestion does only work if we don't want to sort alphabetically

Reply:
what do you think of `findCategories2 () {
    const filterText = this.state.filterText.toLowerCase()

    const categoriesWithTitle = this.props.categories.toArray()
      .filter(category => category.title.toLowerCase().includes(filterText))
      .sort((category1, category2) => category1.title < category2.title ? -1 : category1.title > category2.title ? 1 : 0)

    const categoriesWithContent = this.props.categories.toArray()
      .filter(category => !category.title.toLowerCase().includes(filterText))
      .filter(category => category.content.toLowerCase().includes(filterText))
      .sort((category1, category2) => category1.title < category2.title ? -1 : category1.title > category2.title ? 1 : 0)

    return categoriesWithTitle
      .concat(categoriesWithContent)
      .map(category => ({model: category, children: []}))
  }`?

Reply:
what do you think of
 `findCategories () {
    const filterText = this.state.filterText.toLowerCase()

    const categoriesWithTitle = this.props.categories.toArray()
      .filter(category => category.title.toLowerCase().includes(filterText))
      .sort((category1, category2) => category1.title < category2.title ? -1 : category1.title > category2.title ? 1 : 0)

    const categoriesWithContent = this.props.categories.toArray()
      .filter(category => !category.title.toLowerCase().includes(filterText))
      .filter(category => category.content.toLowerCase().includes(filterText))
      .sort((category1, category2) => category1.title < category2.title ? -1 : category1.title > category2.title ? 1 : 0)

    return categoriesWithTitle
      .concat(categoriesWithContent)
      .map(category => ({model: category, children: []}))
  }`?

Reply:
with some comments :D

Reply:
My solution can be easily adjusted: 
```js
findCategories () {
  return this.props.categories.toArray()
    .map(category => ({priority: prioritizeCategory(category), category})
    .filter(({priority, category}) => priority !== 0)
    .sort((wrapper1, wrapper2) => {
      if(wrapper1.priority !== wrapper2.priority) {
         return wrapper1.priority - wrapper2.priority)
      } else {
        return wrapper1.category.title.localeCompare(wrapper2.category.title)
      }
     })
    .map(wrapper => ({model: wrapper.category, children: []))
}
```
https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

Reply:
I'm fine with both :) Let the PR-OP decide. 

I'd propose to da a function which compares two strings and ignores case.

Reply:
I don't really care which one we use, both is not beautiful :D I like the localeCompare though, so we have now: ```findCategories () {
    const filterText = this.state.filterText.toLowerCase()

    const categoriesWithTitle = this.props.categories.toArray()
      .filter(category => category.title.toLowerCase().includes(filterText))
      .sort((category1, category2) => category1.title.localeCompare(category2.title))

    const categoriesWithContent = this.props.categories.toArray()
      .filter(category => !category.title.toLowerCase().includes(filterText))
      .filter(category => category.content.toLowerCase().includes(filterText))
      .sort((category1, category2) => category1.title.localeCompare(category2.title))

    return categoriesWithTitle
      .concat(categoriesWithContent)
      .map(category => ({model: category, children: []}))
  }``` or @Schedulaar 's suggestion, @NKlug your opinion?

Reply:
I like @klinzo 's code more since it's easier to read and to understand.